moxin-protocol/src/protocol.rs

```diff
@@ -18,4 +129,31 @@ pub enum Response {
 
     // Response to the SearchModels command
     ModelsSearchResults(Vec<Model>),
+
+    // Responses related with the DownloadFile command
+    FileDownloadProgress(FileID, f32), // Progress value from 0.0 to 1.0
+    FileDownloaded(File), // The downloaded_path field is filled
+
+    LoadModelProgress(FileID, f32), // Progress value from 0.0 to 1.0
```

Comment:
Do we currently have to provide progress during the model loading process, or is it sufficient to have a message when loading is complete? Because currently, we are unable to provide progress during the model loading process.

Reply:
The idea, I understand, is to try to implement everything we see in LM Studio. This is something that they implemented.

I will make a note to discuss it in our next weekly call, but yeah.. let's keep this requirement for the future.